packages/swingset/src/types.ts

```diff
@@ -40,3 +40,25 @@ export type EvaluatedEntity<T extends Entity = ComponentEntity | DocsEntity> =
   T & {
     load: () => Promise<React.ElementType>
   }
+
+export type ComponentNode = Pick<
+  ComponentEntity,
+  '__type' | 'title' | 'slug' | 'componentPath' | 'children'
+>
+
+export type FolderNode = {
+  __type: 'folder'
+  title: string
+  parentCategory: string
+  children: ComponentNode[]
+}
+
+export type NavigationNode = ComponentNode | FolderNode
+
+export type CategoryNode = {
+  type: 'category'
```

Comment:
Is `__type` the name we want to use? (I think yes based on your other comments)

```suggestion
  __type: 'category'
```

Reply:
Originally the `Omit<...> & {type: '....'}` changed it but I saw in the first review you requested to remove the Omit.

I'm impartial on the naming but think `type` felt more consistent with the other properties.

Reply:
The other two nodes use `__type`, so I'd say let's pick one or the other and be consistent between each interface 👍

Reply:
Will stick with what's already there then, since `__type` is used across a few other files already